exporters/otlp/otlptrace/internal/otlpconfig/options.go

```diff
@@ -244,11 +248,37 @@ func NewGRPCOption(fn func(cfg Config) Config) GRPCOption {
 
 func WithEndpoint(endpoint string) GenericOption {
 	return newGenericOption(func(cfg Config) Config {
-		cfg.Traces.Endpoint = endpoint
+		// Add scheme if not present
+		if !hasScheme(endpoint) {
+			endpoint = (&url.URL{Scheme: getScheme(cfg), Host: endpoint}).String()
+		}
+		u, err := url.Parse(endpoint)
+		if err != nil {
+			global.Error(err, "parse url", "input", endpoint)
+			return cfg
+		}
+		cfg.Traces.Endpoint = u.Host
+		// For OTLP/HTTP endpoint URLs without a per-signal
+		// configuration, the passed endpoint is used as a base URL
+		// and the signals are sent to these paths relative to that.
+		cfg.Traces.URLPath = path.Join(u.Path, DefaultTracesPath)
 		return cfg
 	})
 }
 
+// getScheme fetches scheme based on config.
+func getScheme(cfg Config) string {
+	if cfg.Traces.Insecure {
+		return "http"
+	}
+	return "https"
+}
+
+// hasScheme determines if the given URL starts with a scheme.
+func hasScheme(rawURL string) bool {
+	return strings.HasPrefix(strings.ToLower(rawURL), "http://") || strings.HasPrefix(strings.ToLower(rawURL), "https://")
```

Comment:
Nit, but we don't need to make the url lower twice.

```suggestion
	rawURL = strings.ToLower(rawURL)
	return strings.HasPrefix(rawURL, "http://") || strings.HasPrefix(rawURL, "https://")
```